Backpack/src/main/java/net/skyscanner/backpack/calendar/view/BpkCalendarLabelsViewModel.kt

```diff
@@ -0,0 +1,113 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.calendar.view
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.text.TextPaint
+import android.text.TextUtils
+import androidx.core.content.ContextCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.calendar.model.CalendarLabel
+import net.skyscanner.backpack.text.BpkText
+import org.threeten.bp.LocalDate
+
+internal class BpkCalendarLabelsViewModel(
```

Comment:
The "renderer" of the labels. Manages and updates all the necessary information, preparing the data set to be drawn.